engine/Pointer.js

```diff
@@ -0,0 +1,43 @@
+import {Path} from 'path.js';
+
+export class Pointer{
+	constructor(container, index){
+		this.container = container;
+		this.index = index;
+	}
+
+	Resolve(){
+		if (index < 0) return this.container;
```

Comment:
We probably want `this.index` here :)

edit: same thing on the next couple of lines